go/pkg/apis/paddlepaddle/v1alpha1/types.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	// CRDKind original TPR name is TraingingJob, but I think
+	// it is too common and easy to conflict
+	CRDKind       = "PdJob"
+	CRDKindPlural = "pdjobs"
+	CRDGroup      = "paddlepaddle.org"
+	CRDVersion    = "v1alpha1"
+)
+
+// CRDName returns name of crd
+func CRDName() string {
+	return fmt.Sprintf("%s.%s", CRDKindPlural, CRDGroup)
+}
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
`k8s:deepcopy-gen` is a code-generation mark, this should be placed above of the struct definition, and you can move https://github.com/PaddlePaddle/cloud/blob/develop/go/api/deepcopy_gen.sh to here to generate code for deep copy.

Reply:
@typhoonzero yes, there is a shell to generate deepcopy,client,informer,lister for the resource, and I did not commit it this time